### PR TITLE
Properties 2.0: New api. Not ready to merge.

### PR DESCRIPTION
**Pull request by baev**
#### PropertyLoader

* Remove all static shit
* New fluent-api builder

    ```java
PropertyLoader.newInstance()
        .withPropertyProvider(provider)
        .withFieldDecorator(decorator)
        .withClassLoader(classLoader)
        .withDefaults(defaults)
        .register(new MyTypeConverter(), MyType.class)
        .populate(this);
    ```
* Annotations removed: `@Use` and `@With`
* Add auto convert for enums
* Throw `PropertyLoaderException` if some convert errors occurs 

#### PropertyProviders

```java
public interface PropertyProvider {
    Properties provide(ClassLoader classLoader, Object bean);
}
```

#### FieldDecorator

```java
public interface FieldDecorator {

    /**
     * Returns true if given field should be decorated,
     * false otherwise.
     */
    boolean shouldDecorate(Field field);

    /**
     * Get field id by given {@link Field}. This method calls <b>only</b>
     * if {@link #shouldDecorate(Field)} method for the same field returns true.
     *
     * @return field id.
     */
    String getFieldKey(Field field);
}
```

### What else I am going to do

 - [x] Change packages to `ru.qatools.properties`
 - [x] Change version to `2.0`
 - [ ] Add ability to dump/load resolved config

### CODE REVIEW

**Comment by baev** [comment]
Summon @lanwen @artkoshelev @eroshenkoam

File: properties-loader/src/test/resources/supported-types.properties [hunk 1]
```diff
@@ -11,4 +11,3 @@ supported.url=http://www.yandex.ru
 supported.uri=file:///var
 supported.enum=mordor
 supported.charset=UTF-8
-supported.invalid.charset=asdasd
```
Comment (lanwen) [inline properties-loader/src/test/resources/supported-types.properties]: why deleted?

Comment (baev) [inline properties-loader/src/test/resources/supported-types.properties]: Now property loader will throw an exception if some convertion errors occurs

Comment (lanwen) [inline properties-loader/src/test/resources/supported-types.properties]: do you have test for it?

Comment (baev) [inline properties-loader/src/test/resources/supported-types.properties]: @lanwen this case covered via `IncorrectValuesOverridingTest`

**Comment by artkoshelev** [comment]
please configure this bot not to spam same messages

**Comment by baev** [comment]
@coveralls could you please shut up and stop spam?

**Comment by lanwen** [comment]
rm one build config from travis - now will be one msg per update

**Pull request merged**